### ISSUE TITLE
Method to close the custom tab?

### ISSUE THREAD

**Issue by hiddencaliber** [droibit/flutter_custom_tabs#17]

**Comment by ashkryab** [comment]
me to

**Comment by ashkryab** [comment]
need close tab after hendle deep link

**Comment by agarzas** [comment]
@ashkryab @hiddencaliber did you find a solution to this issue?

**Comment by yokristiawan-gmail-com** [comment]
If the website is yours, you can use JavaScript window.close() command to close the browser.

**Comment by daniellampl** [comment]
I agree with @ashkryab. After deep linking into the app from a custom tab, the previous custom tab seems to still open in the background. This results in the behavior, that no other custom tab can get opened while the app is running.

@droibit is it possible to close previously opened custom tabs?

**Comment by droibit** [comment]
@daniellampl   
Please consider `url_lancher` to open a deep link URL, 
as I can not provide a good solution from this plugin for now.

I understand that `flutter_custom_tabs` and deep links are incompatible,   
so I'll consider a fundamental solution.

**Comment by ayechanmaung47** [comment]
> @daniellampl Please consider `url_lancher` to open a deep link URL, as I can not provide a good solution from this plugin for now.
> 
> I understand that `flutter_custom_tabs` and deep links are incompatible, so I'll consider a fundamental solution.

deep links work for Android but not for iOS.

**Comment by danwild** [comment]
Dupe of https://github.com/droibit/flutter_custom_tabs/issues/33 ?

Nice work on this plugin @droibit , thanks 👍

But also, this is a critical issue. 

Are there any plans to merge in something like the functionality [mentioned previously](https://github.com/droibit/flutter_custom_tabs/issues/33#issuecomment-942075288)?

Or something similar to [closeInAppWebView](https://pub.dev/documentation/url_launcher/latest/url_launcher/closeInAppWebView.html)?

**Comment by danwild** [comment]
There _is_ an existing method to do this! 😅

Just call [closeCustomTabs](https://github.com/droibit/flutter_custom_tabs/blob/0bd39a33a553540eea4a5747b17089219d4e0aa1/flutter_custom_tabs/lib/src/launcher.dart#L81).

Thanks @droibit 🎉

This kinda addresses https://github.com/droibit/flutter_custom_tabs/issues/33 too

**Comment by droibit** [comment]
Hi,

Thank you for your comments and insights. As you mentioned,   
`flutter_custom_tabs` currently offers the closeCustomTabs function to manually close Custom Tabs.

I'm glad to hear that this function may match your use case. If you are considering using OAuth,   
you may want to explore plugins like [flutter_web_auth_2](https://pub.dev/packages/flutter_web_auth_2) for a better user experience as it leverages OS capabilities.

For now, there are no immediate plans to add additional functionality beyond manually closing Custom Tabs.   
However, I recognize that closing Custom Tabs is a key use case for OAuth,   
and I will begin considering support for OAuth authentication in future versions of the plugin.

Thank you for your feedback and suggestions.
